### ISSUE TITLE
[a11y]: DatePicker Range WCAG 2.4.11 requirement

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19284]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

N/A

### Description

When moving focus via keyboard from a DatePicker element in range mode, the calendar closes after moving focus by pressing TAB but not when moving backwards with SHIFT+TAB. In some cases this will obscure the element in focus and fail WCAG 2.4.11 (Focus not obscured):

![Image](https://github.com/user-attachments/assets/fd56cca7-0775-40ff-9ec1-cccea0096213)

### WCAG 2.1 Violation

WCAG 2.4.11

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datepicker--range-with-calendar-with-layer

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-datepicker--range-with-calendar-with-layer
2. Click on or TAB to focus on the start date input in Layer 3 so that its calendar opens
3. Press Shift+Tab to change focus to the previous datepicker element
4. The datepicker element in focus is obscured by the previous datepicker's calendar

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
